src/argilla_server/contexts/questions.py

```diff
@@ -120,6 +120,10 @@ def _validate_span_question_settings_before_create(
     if field not in field_names:
         raise ValueError(f"'{field}' is not a valid field name.\nValid field names are {field_names!r}")
 
+    for question in dataset.questions:
+        if question.type == QuestionType.span and field == question.parsed_settings.field:
```

Comment:
I already added in one commit a property to `Question` database model to check if if span or not. But after some changes I removed it because I was not using it.

Take a look to the commit here: https://github.com/argilla-io/argilla-server/pull/51/commits/799743b76bad518996ba9fd176d705b9cbe25dd7#diff-b7efe64daa88a5d7c98088e2a66d38739605226e60d4ed967712ca2790c54931

If you want to apply that change then you can do:

```suggestion
        if question.is_span and field == question.parsed_settings.field:
```